src/platforms/posix/drivers/gpssim/gpssim.cpp

```diff
@@ -294,6 +306,16 @@ GPSSIM::receive(int timeout)
 
 		timestamp_last = gps.timestamp;
 
+		// check for data set by the user
+		_report_gps_pos.fix_type = (_fix_type < 0) ? _report_gps_pos.fix_type : _fix_type;
+		_report_gps_pos.satellites_used = (_num_sat < 0) ? _report_gps_pos.satellites_used : _num_sat;
+		// use normal distribution for noise
+		std::normal_distribution<float> normal_distribution(0.0f, 1.0f);
+		std::random_device rd;
+		std::default_random_engine gen(rd());
```

Comment:
`gen` and `rd` need to be class attributes, otherwise it could be that you always get the same random number.

Reply:
done